src/VisualStudio/Core/Impl/CodeModel/NodeKeyValidation.cs

```diff
@@ -17,14 +17,17 @@ public NodeKeyValidation()
 
         public void AddProject(AbstractProject project)
         {
-            var provider = (IProjectCodeModelProvider)project;
-            var fcms = provider.ProjectCodeModel.GetCachedFileCodeModelInstances();
-
-            foreach (var fcm in fcms)
+            var provider = project as IProjectCodeModelProvider;
+            if (provider != null)
```

Comment:
it means it just doesn't participate in codemodely things.  i.e. you can't get the codemodel view of the files in this project.  TypeScript is such an example.  We simply do not participate in CodeModel at all (and have no intent to for the forseeable future).